R/utilities-sensitivity-analysis.R

```diff
@@ -1,342 +1,343 @@
+      numberOfCores = numberOfCores,
+      resultsFileFolder = resultsFileFolder,
+      resultsFileName = resultsFileName
+    )
+  }
+  return(allResultsFileNames)
+}
+
+
+#' @title individualSensitivityAnalysis
+#' @description Run SA for an individual, possibly after modifying the simulation using individualParameters.  Determine whether to run SA for on single core or in parallel.
+#' If on single core, pass simulation to analyzeCoreSensitivity.  If in parallel, pass simulation to runParallelSensitivityAnalysis.
+#' @param simFilePath path to simulation file
+#' @param parametersToPerturb paths to parameters to perturb in sensitivity analysis
+#' @param individualParameters is an object storing an individual's parameters, obtained from a population object's getParameterValuesForIndividual() function.
+#' @param numberOfCores is the number of cores over which to parallelize the sensitivity analysis
+#' @param resultsFileFolder path to population sensitivity analysis results CSV files
+#' @param resultsFileName root name of population sensitivity analysis results CSV files
+#' @return allResultsFileNames, the paths to CSV files containing results of sensitivity analysis
+#' @export
+#' @import ospsuite
+individualSensitivityAnalysis <- function(simFilePath,
+                                          parametersToPerturb = NULL,
+                                          individualParameters,
+                                          numberOfCores = 1,
+                                          resultsFileFolder = resultsFileFolder,
+                                          resultsFileName = resultsFileName) {
+  # Load simulation to determine number of perturbation parameters
+  sim <- loadSimulation(simFilePath)
+
+  # If no perturbation parameters specified, perturb all parameters
+  if (is.null(parametersToPerturb)) {
+    parametersToPerturb <- ospsuite::potentialVariableParameterPathsFor(simulation = sim)
+  }
+  totalNumberParameters <- length(parametersToPerturb)
+
+  # In case there are more cores specified in numberOfCores than there are parameters, ensure at least one parameter per spawned core
+  numberOfCores <- min(numberOfCores, totalNumberParameters)
+  if (totalNumberParameters == 0) {
+    stop("No variable parameters found for sensitivity analysis.")
+  }
+
+  # Determine if SA is to be done on a single core or more
+  if (numberOfCores > 1) {
+    allResultsFileNames <- runParallelSensitivityAnalysis(
+      simFilePath,
+      parametersToPerturb,
+      individualParameters,
+      numberOfCores,
+      resultsFileFolder,
+      resultsFileName
+    )
+  } else {
+    # No parallelization
+    updateSimulationIndividualParameters(simulation = sim, individualParameters)
+    allResultsFileNames <- file.path(resultsFileFolder, paste0(resultsFileName, ".csv"))
+    analyzeCoreSensitivity(
+      simulation = sim,
+      parametersToPerturb = parametersToPerturb,
+      totalSensitivityThreshold = 1,
```

Comment:
That should be  a parameter

Reply:
removing totalSensitivityThreshold for now, adding variationRange instead